wallet-api/src/Wallet/Emulator/Types.hs

```diff
@@ -151,24 +152,25 @@ newtype EmulatedWalletApi a = EmulatedWalletApi { runEmulatedWalletApi :: (Excep
 handleNotifications :: [Notification] -> EmulatedWalletApi ()
 handleNotifications = mapM_ (updateState >=> runTriggers)  where
     updateState = \case
-            BlockHeight h -> modify (blockHeight .~ h)
-            BlockValidated blck -> mapM_ (modify . update) blck >> modify (blockHeight %~ succ)
+            BlockHeight h -> modify (walletBlockHeight .~ h)
+            BlockValidated blck -> mapM_ (modify . update) blck >> modify (walletBlockHeight %~ succ)
 
     runTriggers _ = do
-        h <- gets (view blockHeight)
+        h <- gets (view walletBlockHeight)
         adrs <- gets (view addressMap)
         trg <- gets (view triggers)
 
         let values = AM.values adrs
+            annotate = annTruthValue h values
 
-        -- run a blockchain action with the unspent outputs that
-        -- are relevant to its trigger condition
-        let runTrigger trigger action =
-                runBlockchainAction action h (AM.restrict adrs (Set.fromList $ addresses trigger))
+        let runIfTrue annotTr action =
+                if getAnnot annotTr
+                then runEventHandler action annotTr
+                else pure ()
```

Comment:
Should this be using `checkTrigger`?

Reply:
`checkTrigger` is defined as `getAnnot . annTruthValue h mp` so it does the same thing (except it would compute `annTruthValue` again). I added a comment to clarify